services-js/registry-certs/client/birth/components/VerifyIdentificationComponent.tsx

```diff
@@ -1,20 +1,224 @@
 import React from 'react';
 
-interface Props {}
+import { css } from 'emotion';
+
+import {
+  Radio,
+  UploadPhoto,
+  BLACK,
+  OPTIMISTIC_BLUE_LIGHT,
+} from '@cityofboston/react-fleet';
+
+import FieldsetComponent from './FieldsetComponent';
+import SupportingDocumentsInput from './SupportingDocumentsInput';
+import IdIcon from '../icons/IdIcon';
+
+import { SECTION_HEADING_STYLING, SUPPORTING_TEXT_STYLING } from '../styling';
+
+// todo: props to pass addt’l info in if Registry has initiated this
+interface Props {
+  supportingDocuments: (documents: File[]) => void;
+  idImages: (side: string, image: any) => void;
+  isComplete?: (status: boolean) => void;
+}
+
+interface State {
+  requireSupportingDocuments: boolean;
+  hasIdFront: boolean;
+  hasIdBack: boolean;
+  hasDocuments: boolean;
+}
 
 /**
  * Component which allows a user to upload images and supporting files for
  * identification verification. User will also have the ability to take photos
  * with their current device, if possible.
  */
 export default class VerifyIdentificationComponent extends React.Component<
-  Props
+  Props,
+  State
 > {
+  state: State = {
+    requireSupportingDocuments: false,
+    hasIdFront: false,
+    hasIdBack: false,
+    hasDocuments: false,
+  };
+
+  // User cannot proceed before submitting a front and back scan of their ID.
+  // They must also submit supporting documents before proceeding, if those
+  // are required.
+  private checkIsComplete() {
+    const idComplete = this.state.hasIdFront && this.state.hasIdBack;
+
+    if (!this.props.isComplete) {
+      return;
+    }
+
+    if (
+      idComplete &&
+      this.state.requireSupportingDocuments &&
+      this.state.hasDocuments
+    ) {
+      this.props.isComplete(true);
+    } else if (idComplete && !this.state.requireSupportingDocuments) {
+      this.props.isComplete(true);
+    } else {
+      this.props.isComplete(false);
+    }
+  }
+
+  handleFilesChange = (documents: File[]): void => {
+    this.setState({ hasDocuments: !!documents.length });
+
+    this.props.supportingDocuments(documents);
+  };
+
+  handleIdImageChange = (side: string, image: File | null): void => {
+    if (side === 'front') {
+      this.setState({ hasIdFront: !!image });
+    } else if (side === 'back') {
+      this.setState({ hasIdBack: !!image });
+    }
+
+    this.props.idImages(side, image);
+  };
+
+  handleBooleanChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    this.setState({
+      requireSupportingDocuments: event.target.value === 'yes',
+    });
+  };
+
+  // Inform the parent whether or not all required information has been added
+  // by the user, if necessary.
+  componentDidUpdate(_prevProps: Readonly<Props>, prevState: Readonly<State>) {
+    if (this.props.isComplete && prevState !== this.state) {
```

Comment:
the `this.props.isComplete` check here feels like it’s not `componentDidUpdate`’s concern. That can be handled in `checkIsComplete`.

Is it ok that `prevState !== this.state` isn’t a deep equals?

Reply:
Ah, okay - I figured if this component was being used by a parent that didn’t care about `isComplete`, then `checkIsComplete` will never be called... but I guess it doesn’t matter **that** much!

`prevState !== this.state` is fine in this case because we don’t care _what_ is in the state, only that it has changed in order to call `checkIsComplete`.